src/cacheout/cache.py

```diff
@@ -46,6 +47,21 @@ class RemovalCause(Enum):
     POPITEM = auto()
 
 
+#: Callback that will be executed when a cache entry is retrieved.
+
+#: It is called with arguments ``(key, value, exists)`` where `key` is the cache key,
+#: `value` is the value retrieved (could be the default),
+#: and `exists` is whether the cache key exists or not.
+T_ON_GET_CALLBACK = t.Optional[t.Callable[[t.Hashable, t.Any, bool], None]]
+
+#: Callback that will be executed when a cache entry is removed.
+
+#: It is called with arguments ``(key, value, cause)`` where `key` is the cache key,
+#: `value` is the cached value at the time of deletion,
+#: and `cause` is the reason the key was removed (see :class:`RemovalCause` for enumerated causes).
+T_ON_DELETE_CALLBACK = t.Optional[t.Callable[[t.Hashable, t.Any, RemovalCause], None]]
```

Comment:
nit: Constants should all be at the top after the imports. Class definitions next and then function definitions.

Reply:
`T_ON_DELETE_CALLBACK` must be after `RemovalCause` because it depends on `RemovalCause`.